### PR TITLE
Bah 781a Upload Secondary Sources

### PR DESCRIPTION
**Pull request by imeecuison**
## Description
Page for option to enter supporting sources for stressful incident(s).
```bah-781a-upload-sources```
http://localhost:3001/disability-benefits/apply/form-526-all-claims/disabilities/ptsd-secondary-upload-supporting-sources-0

## Testing done
tests/config/781.unit.spec.js

## Screenshots

screen shot 2018-11-27 at 3 29 25 pm

## Acceptance criteria
- [ ] The PTSD type that the user previously selected (either Personal Assault or Military Sexual Trauma) should be populated as the $PTSDclassification on this page, per the Content Document
If the user selected more than one PTSD Type for 0781a (Personal assault and Military sexual trauma) then the following is expected:

- then 'Personal assault & Military sexual trauma' replaces $PTSDclassification in header of

781 workflow

- then 'Personal assault and Military sexual trauma' replaces $PTSDclassification in body of

781 workflow
- [ ]  User should have the option to upload supporting documents per Content Document
If user selects to upload then user should be taken to screen to upload a document

* If user clicks Upload Document button then should be able to select Document type and enter a Document name

* User should have ability to delete a file

* User should be able to add more than one document by clicking 'Add Another'

* Limit on how many docs can be uploaded- TBD

- [ ]  If user selects not to upload then user is taken to the end of the workflow for that incident and asked if there is another incident to enter

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by desireeturner** [comment]
This PR is blocking https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/13104

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 1]
```diff
@@ -105,6 +111,24 @@ export function createFormConfig781a(iterations) {
         uiSchema: secondaryIncidentDate.uiSchema(index),
         schema: secondaryIncidentDate.schema(index),
       },
+      [`secondaryUploadSourcesChoice${index}`]: {
+        title: `${
+          numberToWords[index]
+        } 781a PTSD Upload Supporting Sources Choice`,
+        path: `disabilities/ptsd-secondary-upload-supporting-sources-choice-${index}`,
+        // The Depends will need to be refactored to account for the page index/incident Number
+        depends: isAnswering781aQuestions,
```
Comment (dmccurdybah) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: This needs to be a function call (isAnswering781 returns a function)

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ import { PTSD, PTSD_INCIDENT_ITERATION } from '../constants';
 
 import fullSchema from 'vets-json-schema/dist/21-526EZ-ALLCLAIMS-schema.json';
 
-const formConfig = {
+const createFormConfig = {
```
Comment (dmccurdybah) [inline src/applications/disability-benefits/all-claims/config/form.js]: I think this is a search/replace error

File: src/applications/disability-benefits/all-claims/pages/secondaryUploadSourcesChoice.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
```
Comment (dmccurdybah) [inline src/applications/disability-benefits/all-claims/pages/secondaryUploadSourcesChoice.js]: Can you make this a .jsx file?

File: src/applications/disability-benefits/all-claims/tests/config/781.unit.spec.js [hunk 4]
```diff
@@ -161,5 +161,17 @@ describe('781/781a incident form config iterators', () => {
         ).to.haveOwnProperty('type');
       }
     });
+    it('should return three secondary upload sources page config objects', () => {
+      const config = createFormConfig781a(PTSD_INCIDENT_ITERATION);
+
+      const testMultipleSecondaryUploadSourcesProps = () => {
```
Comment (dmccurdybah) [inline src/applications/disability-benefits/all-claims/tests/config/781.unit.spec.js]: Can you remove the wrapping function here?  I think the test name is clear enough and I've been trying to clean these up

**Review by devin-mccurdy — CHANGES_REQUESTED** [review]

**Review by devin-mccurdy — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 5]
```diff
@@ -564,6 +564,9 @@ export const isUploading781Form = formData =>
 export const isUploading781aForm = formData =>
   _.get('view:upload781aChoice', formData, '') === 'upload';
 
+export const isUploading781aSupportingDocuments = index => formData =>
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/utils.jsx]: Does this have a unit test?

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 6]
```diff
@@ -105,6 +111,22 @@ export function createFormConfig781a(iterations) {
         uiSchema: secondaryIncidentDate.uiSchema(index),
         schema: secondaryIncidentDate.schema(index),
       },
+      [`secondaryUploadSourcesChoice${index}`]: {
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: Does this need a unit test? I see one for `secondaryUploadSources`

**Review by rianfowler — CHANGES_REQUESTED** [review]
Some comments inline.

Can we get a reviewer who has context to indicate the acceptance criteria are met.

this item is also not checked- Is this something that requires more consideration? 
> No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

File: src/applications/disability-benefits/all-claims/config/781/index.js (hunk 6 above)
Comment (zurbergram) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: added unit test

**Review by rianfowler — APPROVED** [review]

**Pull request merged**